native/acter/src/api/room.rs

```diff
@@ -545,27 +545,33 @@ impl SpaceRelations {
 
     pub async fn query_hierarchy(&self) -> Result<Vec<SpaceHierarchyRoomInfo>> {
         let c = self.room.core.clone();
-        let room = self.room.room.clone();
+        let suggested_rooms = self
+            .core
+            .children
+            .iter()
+            .filter(|c| c.suggested())
+            .map(|c| c.room_id())
+            .collect::<Vec<_>>();
         let room_id = self.room.room_id().to_owned();
         RUNTIME
             .spawn(async move {
-                let relations = c.space_relations(&room).await?.children.iter().filter(|c| c.suggested()).map(|c| c.room_id()).collect::<Vec<_>>();
```

Comment:
This is a pure optimization when I realized we didn't have to make that call again but already have it under `self.core`...